### DISCUSSION TITLE
Opacity / Transparency for DrawTexture and DrawImage

### DISCUSSION THREAD

**Discussion by matthijsoudelohuis** [bonsai-rx/bonsai#1602] [Q&A]
Hi there, 

I was wondering if there is already an option to change the opacity/transparency of loaded textures and images in BonVision (or some way around it without directly setting opacity). This would allow merging and combining different images, 
This relates to this (open) issue: 
https://github.com/bonvision/BonVision/issues/23
Thanks for Bonsai/Bonvision and the community :)
Matthijs

**Comment by bruno-f-cruz** [comment]
Hey
Not sure I totally follow. Transparency (alpha) is already supported by the Shaders package. It will just inherit whatever alpha channel the provided image has:
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/1f034445-e824-49c5-8c67-85030fec4aec)

[alphademo.zip](https://github.com/bonsai-rx/bonsai/files/13228591/alphademo.zip)

Can you explain a bit better what you want to achieve?

**Reply by matthijsoudelohuis** [reply]
Hi Bruno, 

That's great, thanks for answering, didn't know that. 
I was working with grayscale .jpg images and the fact that the Shaders package loads the alpha channel if provided in the image is great.
I have played around with storing different RGBA values and storing them as .png. 
This works as expected with 2D rendering with Normalized View and DrawImage as used in your demo script. With my 3D Sphere Mapping setup I also got it to work except that for some reason the transparent-to-opaque range of alpha values is effectively 0-64 (64 being opaque) rather than 0-256. 

![transparentCorridor](https://github.com/bonsai-rx/bonsai/assets/24936970/0c16aeab-3246-4901-baac-5f692b8e9a86)

I am now bumping into another minor problem, which is that I used to load 100 .jpg images with different settings as an ImageSequence and index them. This saves a lot of manual entering of filenames for each transparency level. However, I can not get this to work with my series of .png images (loading each individually works). If you also have any suggestion on this one that'd be great. 
Attached is the workflow with cubemap rendering and some texture A with 100 transparency levels.

Cheers, Matthijs

[alphacorridor.zip](https://github.com/bonsai-rx/bonsai/files/13241103/alphacorridor.zip)
